modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/SpreadSensitivityCalculator.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.credit;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.market.curve.CurveInfoType;
+import com.opengamma.strata.market.curve.CurveName;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivity;
+import com.opengamma.strata.product.credit.ResolvedCds;
+import com.opengamma.strata.product.credit.ResolvedCdsIndexTrade;
+import com.opengamma.strata.product.credit.ResolvedCdsTrade;
+
+/**
+ * The spread sensitivity calculator. 
+ * <p>
+ * The spread sensitivity, also called CS01, is the sensitivity of the CDS product present value to par spreads of the bucket CDSs. 
+ * The bucket CDSs do not necessarily correspond to the node point of the input credit curve.
+ */
+abstract class SpreadSensitivityCalculator {
```

Comment:
As this class is package scoped, the public methods cannot be called unless they are overridden. Was that intended?